src/storage/accessors/FileDataAccessor.ts

```diff
@@ -159,8 +160,14 @@ export class FileDataAccessor implements DataAccessor {
    */
   private async getFileMetadata(link: ResourceLink, stats: Stats):
   Promise<RepresentationMetadata> {
-    return (await this.getBaseMetadata(link, stats, false))
-      .set(CONTENT_TYPE_TERM, link.contentType);
+    const metadata = await this.getBaseMetadata(link, stats, false);
+    // If the resource is using an unsupported contentType, the original contentType was written to the metadata file.
+    // As a result, we should only set the contentType derived from the file path,
+    // when no previous metadata entry for contentType is present.
+    if (link.isMetadata || metadata.contentType === undefined) {
```

Comment:
I don't think it can actually ever occur in practice that `getFileMetadata` gets called with `link` pointing to a metadata resource. Is this something you noticed during integration tests or is this just for completeness?

Reply:
At some point there were tests failing, but this was probably due to how I initially implemented the changes in `writeMetadata`. I executed the tests again without this condition and they all pass now.